pkgs/development/tools/build-managers/bazel/bin_to_env.patch

```diff
@@ -0,0 +1,207 @@
+diff -ur a/src/main/java/com/google/devtools/build/lib/analysis/CommandHelper.java b/src/main/java/com/google/devtools/build/lib/analysis/CommandHelper.java
+--- a/src/main/java/com/google/devtools/build/lib/analysis/CommandHelper.java	1980-01-01 00:00:00.000000000 -0800
++++ b/src/main/java/com/google/devtools/build/lib/analysis/CommandHelper.java	2017-02-19 15:37:06.021759347 -0800
+@@ -242,7 +242,7 @@
+   private static Artifact buildCommandLineArtifact(RuleContext ruleContext, String command,
+       String scriptPostFix) {
+     String scriptFileName = ruleContext.getTarget().getName() + scriptPostFix;
+-    String scriptFileContents = "#!/bin/bash\n" + command;
++    String scriptFileContents = "#!/usr/bin/env bash\n" + command;
+     Artifact scriptFileArtifact = FileWriteAction.createFile(
+         ruleContext, scriptFileName, scriptFileContents, /*executable=*/true);
+     return scriptFileArtifact;
+diff -ur a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java
+--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java	1980-01-01 00:00:00.000000000 -0800
++++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java	2017-02-19 15:37:06.022759356 -0800
+@@ -73,7 +73,7 @@
+         return new PathFragment("/usr/local/bin/bash");
+       }
+     }
+-    return new PathFragment("/bin/bash");
++    return new PathFragment("bash");
+   }
+ 
+   @Override
+diff -ur a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt
+--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt	1980-01-01 00:00:00.000000000 -0800
++++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt	2017-02-19 15:37:06.022759356 -0800
+@@ -1,4 +1,4 @@
+-#!/bin/bash --posix
++#!/usr/bin/env bash
+ # Copyright 2014 The Bazel Authors. All rights reserved.
+ #
+ # Licensed under the Apache License, Version 2.0 (the "License");
+diff -ur a/src/test/java/com/google/devtools/build/lib/shell/CommandTest.java b/src/test/java/com/google/devtools/build/lib/shell/CommandTest.java
+--- a/src/test/java/com/google/devtools/build/lib/shell/CommandTest.java	1980-01-01 00:00:00.000000000 -0800
++++ b/src/test/java/com/google/devtools/build/lib/shell/CommandTest.java	2017-02-19 15:37:06.022759356 -0800
+@@ -70,7 +70,7 @@
+     }
+ 
+     try {
+-      new Command(new String[] {"/bin/true", null}).execute();
++      new Command(new String[] {"/usr/bin/env", "true", null}).execute();
```

Comment:
imo this ought to stay, since it looks like it won't work without the patch.  In theory you could patch in the /nix/store path to bin/true but if it's findable in $PATH then this is probably good enough